src/imitation/scripts/config/eval_policy.py

```diff
@@ -37,88 +42,89 @@ def explore_eps_greedy():
 
 @eval_policy_ex.named_config
 def render():
-    common = dict(num_vec=1, parallel=False)
+    environment = dict(num_vec=1, parallel=False)
     render = True
 
 
 @eval_policy_ex.named_config
 def acrobot():
-    common = dict(env_name="Acrobot-v1")
+    environment_name = dict(gym_id="Acrobot-v1")
 
 
 @eval_policy_ex.named_config
 def ant():
-    common = dict(env_name="Ant-v2")
+    environment_name = dict(gym_id="Ant-v2")
 
 
 @eval_policy_ex.named_config
 def cartpole():
-    common = dict(env_name="CartPole-v1")
+    environment_name = dict(gym_id="CartPole-v1")
 
 
 @eval_policy_ex.named_config
 def seals_cartpole():
-    common = dict(env_name="seals/CartPole-v0")
+    environment_name = dict(gym_id="seals/CartPole-v0")
 
 
 @eval_policy_ex.named_config
 def half_cheetah():
-    common = dict(env_name="HalfCheetah-v2")
+    environment_name = dict(gym_id="HalfCheetah-v2")
 
 
 @eval_policy_ex.named_config
 def seals_half_cheetah():
-    common = dict(env_name="seals/HalfCheetah-v0")
+    environment_name = dict(gym_id="seals/HalfCheetah-v0")
 
 
 @eval_policy_ex.named_config
 def seals_hopper():
-    common = dict(env_name="seals/Hopper-v0")
+    environment_name = dict(gym_id="seals/Hopper-v0")
 
 
 @eval_policy_ex.named_config
 def seals_humanoid():
-    common = dict(env_name="seals/Humanoid-v0")
+    environment_name = dict(gym_id="seals/Humanoid-v0")
 
 
 @eval_policy_ex.named_config
 def mountain_car():
-    common = dict(env_name="MountainCar-v0")
+    environment_name = dict(gym_id="MountainCar-v0")
 
 
 @eval_policy_ex.named_config
 def seals_mountain_car():
-    common = dict(env_name="seals/MountainCar-v0")
+    environment_name = dict(gym_id="seals/MountainCar-v0")
 
 
 @eval_policy_ex.named_config
 def pendulum():
-    common = dict(env_name="Pendulum-v1")
+    environment_name = dict(gym_id="Pendulum-v1")
 
 
 @eval_policy_ex.named_config
 def reacher():
-    common = dict(env_name="Reacher-v2")
+    environment_name = dict(gym_id="Reacher-v2")
 
 
 @eval_policy_ex.named_config
 def seals_ant():
-    common = dict(env_name="seals/Ant-v0")
+    environment_name = dict(gym_id="seals/Ant-v0")
 
 
 @eval_policy_ex.named_config
 def seals_swimmer():
-    common = dict(env_name="seals/Swimmer-v0")
+    environment_name = dict(gym_id="seals/Swimmer-v0")
 
 
 @eval_policy_ex.named_config
 def seals_walker():
-    common = dict(env_name="seals/Walker2d-v0")
+    environment_name = dict(gym_id="seals/Walker2d-v0")
 
 
 @eval_policy_ex.named_config
 def fast():
-    common = dict(env_name="seals/CartPole-v0", num_vec=1, parallel=False)
+    environment_name = dict(gym_id="seals/CartPole-v0")
```

Comment:
This was not a problem before because we had a `common` ingredient that we used as kind of a "catch all" ingredient. Up to now that was for logging, for the random number generator and for the environment.

I decided to break this common ingredient apart into smaller/simpler ingredients for a number of reasons:

1. I found the experiment/ingredient analogy of sacred rather helpful. I was often thinking a physical lab experiment to visualize what was going on. The `common` ingredient did not really fit into this analogy.
2. When I read a package name like "common" I have no idea what to expect inside. I have to open and read the package. This makes exploring the library and understanding it's architecture really hard for newcomers. I think the current split makes the library more readable in general.
3. I was afraid that such a "catch all" ingredient might grow over time since it is so convenient to add new stuff to something that has no clearly defined scope after all. It was already moving towards becoming a [god object](https://en.wikipedia.org/wiki/God_object)

Does this makes things clear @Rocamonde ?

Reply:
AFAICT you can merge environment and environment_name into the same ingredient, and remove the dependency of logger on environment_name. You only use this on the hooks, at which point the config hierarchy has already loaded, and you can read the environment_name off the config object anyways, assuming that this ingredient was used. All our experiments use the environment ingredient, but just to make things type safe, you could either throw an error, or use a default using `.get()`.

Reply:
Thanks for the hint. I will look into this!

Reply:
When I remove the `environment_name_ingredient` from the `logging_ingredient` some (but not all) script tests fail because of a `KeyError` when accessing the `config["environment_name"]`. So I guess it depends on the `environment_name` being initialized before or after the `logging_ingredient`. So unfortunately I see no way to merge the two ingredients. If you have other ideas I am open to them!

Reply:
You can't just remove the environment_name ingredient from the logging; you need to add the key to the environment ingredient. Is this what you tried? (based on your description, it sounds like it might not be). What scripts fail?

Reply:
I went all the way and also tried your suggestion but I think it leads to the same scenario: accessing the config of a foreign ingredient inside a config hook without declaring that foreign ingredient as a dependency.

This are the fails:

```
FAILED tests/scripts/test_scripts.py::test_main_console[imitation.scripts.eval_policy] - SystemExit: 1
FAILED tests/scripts/test_scripts.py::test_main_console[imitation.scripts.train_adversarial] - SystemExit: 1
FAILED tests/scripts/test_scripts.py::test_main_console[imitation.scripts.train_imitation] - SystemExit: 1
FAILED tests/scripts/test_scripts.py::test_main_console[imitation.scripts.train_preference_comparisons] - SystemExit: 1
FAILED tests/scripts/test_scripts.py::test_main_console[imitation.scripts.train_rl] - SystemExit: 1
FAILED tests/scripts/test_scripts.py::test_train_preference_comparisons_main[plain] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_preference_comparisons_main[with_expert_trajectories] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_preference_comparisons_main[warmstart] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_preference_comparisons_main[with_checkpoints] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_preference_comparisons_envs_no_crash[seals_cartpole] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_preference_comparisons_envs_no_crash[mountain_car] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_preference_comparisons_envs_no_crash[seals_mountain_car] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_preference_comparisons_sac - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_preference_comparisons_sac_reward_relabel - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_preference_comparisons_reward_named_config[named_configs0] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_preference_comparisons_reward_named_config[named_configs1] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_preference_comparisons_reward_named_config[named_configs2] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_dagger_main - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_dagger_warmstart - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_bc_main_with_none_demonstrations_raises_value_error - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_bc_main[expert_from_path] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_bc_main[expert_from_huggingface] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_bc_main[random_expert] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_bc_main[zero_expert] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_bc_warmstart - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_rl_main[cold_start] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_rl_main[warm_start] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_rl_wb_logging - AssertionError: Regex pattern '.*api_key not configured.*' does not match "'environment'".
FAILED tests/scripts/test_scripts.py::test_train_rl_sac - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_eval_policy[config0] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_eval_policy[config1] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_eval_policy[config2] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_eval_policy[config3] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_eval_policy[config4] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_adversarial[airl-named_configs0] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_adversarial[airl-named_configs1] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_adversarial[airl-named_configs2] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_adversarial[gail-named_configs0] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_adversarial[gail-named_configs1] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_adversarial[gail-named_configs2] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_adversarial_warmstart[airl] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_adversarial_warmstart[gail] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_adversarial_sac[airl] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_adversarial_sac[gail] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_train_adversarial_algorithm_value_error - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_parallel[config_updates0] - ray.tune.error.TuneError: ('Trials did not complete', [inner_de9f4_00000, inner_de9f4_00001, inner_de9f4_00002, inner_de9f4_00003])
FAILED tests/scripts/test_scripts.py::test_parallel[config_updates1] - ray.tune.error.TuneError: ('Trials did not complete', [inner_fb2be_00000, inner_fb2be_00001, inner_fb2be_00002, inner_fb2be_00003, inner_fb2be_...
FAILED tests/scripts/test_scripts.py::test_parallel_train_adversarial_custom_env - ray.tune.error.TuneError: ('Trials did not complete', [inner_18e52_00000])
FAILED tests/scripts/test_scripts.py::test_analyze_imitation[_run_train_adv_for_test_analyze_imit-run_names1] - KeyError: 'environment'
FAILED tests/scripts/test_scripts.py::test_analyze_gather_tb - ray.tune.error.TuneError: ('Trials did not complete', [inner_21f65_00000, inner_21f65_00001])
```